### DISCUSSION TITLE
Background Image/Color for PDFs

### DISCUSSION THREAD

**Discussion by shuvashish76** [koreader/koreader#11575] [Ideas]
[Librera](https://github.com/foobnix/LibreraReader) has the feature to change background.
![Screenshot_2024-03-19_20-42-43_1](https://github.com/koreader/koreader/assets/48226312/08fd2315-1954-48a5-92ab-a726c5a914cf)

How it looks on KOReader:
<details><summary>Screenshot</summary>
<p>

![Screenshot_20240319-205057_KOReader](https://github.com/koreader/koreader/assets/48226312/3697efa7-4534-48fc-87c6-7f58dc9fc8b0)

</p>
</details> 

For Librera:
<details><summary>Screenshot</summary>
<p>

![Screenshot_20240319-205226_Librera_FD](https://github.com/koreader/koreader/assets/48226312/d10d6d0b-010c-4eed-84d6-c4da29a47fb5)

</p>
</details> 

---

1. When I try dark mode on KOReader the colors of images are inverted so I'm forced to use light theme. For Librera the background color doesn't change the PDF images significantly.
2. Both app use MuPDF engine so I think it should be possible for KOReader as well.
3. Another option for me is to use "Blue Light filter" apps but still the changes are noticeable. Using inbuilt background better than 3rd party workarounds.
4. I understand KOReader mostly aimed at eInk devices not colored devices (e.g phone or desktop), so this feature is not in priority (possibly never will be), [QUESTION] can we at least use dark mode without inverting _images_ color somehow?

**Comment by Frenzie** [comment]
At a glance it's not using MuPDF but doing some kind of image manipulation on it by using white as transparency. (But that's just a quick skim; I could be misreading it.)
https://github.com/foobnix/LibreraReader/blob/86522dc1c73cb70e2eb9f64e2bc46aae6254d666/app/src/main/java/org/ebookdroid/core/EventDraw.java#L99-L153

**Reply by pazos** [reply]
Yay. Alpha blending.

Not fun on cpu :)

**Reply by Frenzie** [reply]
To be fair if you could set a different background in MuPDF it shouldn't be terribly different from what it does in the first place.

**Comment by shuvashish76** [comment]
> https://github.com/Euphoriyy/KOReader.patches#-2-ui-background-colorlua

@Euphoriyy 's User patch now supports changing page background color for fixed layout documents.
Please add "user patch available" label.

Edit: Now available as a single plugin: https://github.com/Euphoriyy/appearance.koplugin
